sentinel-2/ndwi/eob.js

```diff
@@ -0,0 +1,43 @@
+//VERSION=3
+//ndwi
+const ramp = [
+    [-0.8, 0x008000],
+    [0, 0xFFFFFF],
+    [0.8, 0x0000CC]
+];
+
+let viz = new ColorRampVisualizer(ramp);
+
+function setup() {
+    return {
+        input: ["B03", "B08", "SCL", "dataMask"],
+        output: [
+            { id: "default", bands: 4 },
+            { id: "index", bands: 1, sampleType: "FLOAT32" },
+            { id: "eobrowserStats", bands: 2, sampleType: 'FLOAT32' },
+            { id: "dataMask", bands: 1 }
+        ]
+    };
+}
+
+function evaluatePixel(samples) {
+    let val = index(samples.B03, samples.B08);
+    // The library for tiffs works well only if there is only one channel returned.
+    // So we encode the "no data" as NaN here and ignore NaNs on frontend.
+    const indexVal = samples.dataMask === 1 ? val : NaN;
+
+    const imgVals = [...viz.process(val), samples.dataMask];
```

Comment:
Let's keep this consistent and use `imgVals = viz.process(val)` and `imgVals.concat(samples.dataMask)`